packages/synapse-react-client/src/components/QueryContext/QueryContext.tsx

```diff
@@ -44,52 +28,52 @@ export type QueryContextType<
+  /** react-query UseQueryOptions to fetch the current page of table data */
+  rowDataQueryOptions: TableQueryUseQueryOptions['rowDataQueryOptions']
+  /** react-query UseInfiniteQueryOptions to fetch infinite rows of table data */
+  rowDataInfiniteQueryOptions: TableQueryUseQueryOptions['rowDataInfiniteQueryOptions']
+  /** react-query UseQueryOptions to fetch the current query metadata (e.g. column models, facet statistics, total number of rows) */
+  queryMetadataQueryOptions: TableQueryUseQueryOptions['queryMetadataQueryOptions']
+
+  /** Navigates to a particular page. The first page has value 1. Note that this has no effect if `isInfinite` is true */
   goToPage: (pageNum: number) => void
-  /** The current page number, where page 1 is the first page. */
+  /** The current page number, where page 1 is the first page. Note that this has no effect if `isInfinite` is true. */
   currentPage: number
   /** Updates the page size */
   setPageSize: (pageSize: number) => void
   /** The current page size. */
   pageSize: number
-}
 
-export type InfiniteQueryContextType<
-  TIncludedFields extends OptionalQueryBundleRequestFields = never,
-  ExcludeOtherFields extends true | false = false,
-> = QueryContextType<TIncludedFields, ExcludeOtherFields> & {
-  /** Returns true when loading a new page of query results */
-  isLoadingNewPage: boolean
-  /** Whether the query result bundle has a next page */
-  hasNextPage?: boolean
-  /** Invoke this method to fetch and append the next page of rows to the data  */
-  appendNextPageToResults: () => Promise<void>
-  /** Invoke to fetch and update the data with the next page of query results */
-  goToNextPage: () => Promise<void>
-  /** Whether the query result bundle has a previous page */
-  hasPreviousPage: boolean
-  /** Invoke to fetch and update the data with the previous page of query results */
-  goToPreviousPage: () => Promise<void>
+  /**
+   * A column name may be "locked" so that it is both (1) not shown to the user that the filter is active, and (2) not modifiable by the user.
+   * For example, we may show only the data matching a particular facet value on a Details Page without implying that the shown data is part of a larger table.
+   * The presence of a locked filter will result in a client-side modification of the active query and result bundle data.
+   */
+  lockedColumn: LockedColumn | undefined
+
+  /**
+   * PORTALS-3071: For Tables that are not entityviews or a datasets, keep track of the column that should be used for the row entity ID
+   */
+  fileIdColumnName: string | undefined
+  /**
+   * PORTALS-3071: For Tables that are not entityviews or a datasets, keep track of the column that should be used for the row (entity) version
+   */
+  fileVersionColumnName: string | undefined
+  /**
+   * PORTALS-3071: For Tables that are not entityviews or a datasets, keep track of the column that should be used for the row (entity) name
+   */
+  fileNameColumnName: string | undefined
```

Comment:
These were all stored in atoms, but never changed, so they didn't benefit from the atomic state pattern. Moved to context.